### PR TITLE
IGNITE-16061 Make BinaryMarshaller#setBinaryContext() public

### PR DESCRIPTION
**Pull request by rpuch**
BinaryMarshaller#setBinaryContext() method is required to initialize BinaryMarshaller, without this method invocation BinaryMarshaller cannot be used. But the method was private and was never invoked from BinaryMarshaller code itself. Instead, it was always called from other classes using reflection, like this:

IgniteUtils.invoke(BinaryMarshaller.class, bMarsh0, "setBinaryContext", binaryCtx, ctx.config());

So, in reality, the method was handled like it was public, but as it was actually private, reflection had to be used as a work-around.

That's why we simply make it public now.

Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/binary/CacheObjectBinaryProcessorImpl.java [hunk 1]
```diff
@@ -34,6 +34,7 @@
 import java.util.concurrent.ConcurrentMap;
 import java.util.function.BooleanSupplier;
 import javax.cache.CacheException;
+
```
Comment (tkalkirill) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/binary/CacheObjectBinaryProcessorImpl.java:37]: Not needed

File: modules/core/src/test/java/org/apache/ignite/internal/binary/BinaryFieldExtractionSelfTest.java [hunk 2]
```diff
@@ -22,10 +22,10 @@
 import java.nio.ByteBuffer;
 import java.sql.Time;
 import java.util.concurrent.ThreadLocalRandom;
+
```
Comment (tkalkirill) [inline modules/core/src/test/java/org/apache/ignite/internal/binary/BinaryFieldExtractionSelfTest.java:25]: Not needed

File: modules/core/src/test/java/org/apache/ignite/internal/binary/BinaryFieldsAbstractSelfTest.java [hunk 3]
```diff
@@ -22,12 +22,12 @@
 import java.util.Arrays;
 import java.util.Date;
 import java.util.UUID;
+
```
Comment (tkalkirill) [inline modules/core/src/test/java/org/apache/ignite/internal/binary/BinaryFieldsAbstractSelfTest.java:25]: Not needed

File: modules/core/src/test/java/org/apache/ignite/internal/binary/BinaryFooterOffsetsAbstractSelfTest.java [hunk 4]
```diff
@@ -18,11 +18,11 @@
 package org.apache.ignite.internal.binary;
 
 import java.util.Arrays;
+
```
Comment (tkalkirill) [inline modules/core/src/test/java/org/apache/ignite/internal/binary/BinaryFooterOffsetsAbstractSelfTest.java:21]: Not needed

File: modules/core/src/test/java/org/apache/ignite/internal/binary/BinaryMarshallerSelfTest.java [hunk 5]
```diff
@@ -56,6 +56,7 @@
 import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
+
```
Comment (tkalkirill) [inline modules/core/src/test/java/org/apache/ignite/internal/binary/BinaryMarshallerSelfTest.java:59]: Not needed

File: modules/core/src/test/java/org/apache/ignite/internal/binary/GridBinaryMarshallerCtxDisabledSelfTest.java [hunk 6]
```diff
@@ -22,13 +22,13 @@
 import java.io.ObjectInput;
 import java.io.ObjectOutput;
 import java.util.Arrays;
+
```
Comment (tkalkirill) [inline modules/core/src/test/java/org/apache/ignite/internal/binary/GridBinaryMarshallerCtxDisabledSelfTest.java:25]: Not needed

File: modules/core/src/test/java/org/apache/ignite/internal/binary/GridBinaryWildcardsSelfTest.java [hunk 7]
```diff
@@ -21,6 +21,7 @@
 import java.util.Collection;
 import java.util.Map;
 import java.util.concurrent.ConcurrentMap;
+
```
Comment (tkalkirill) [inline modules/core/src/test/java/org/apache/ignite/internal/binary/GridBinaryWildcardsSelfTest.java:24]: Not needed

File: modules/core/src/test/java/org/apache/ignite/testframework/junits/GridAbstractTest.java [hunk 8]
```diff
@@ -54,6 +54,7 @@
 import javax.management.MBeanServerInvocationHandler;
 import javax.management.MalformedObjectNameException;
 import javax.management.ObjectName;
+
```
Comment (tkalkirill) [inline modules/core/src/test/java/org/apache/ignite/testframework/junits/GridAbstractTest.java:57]: Not needed

File: modules/core/src/test/java/org/apache/ignite/testframework/junits/IgniteTestResources.java [hunk 9]
```diff
@@ -24,6 +24,7 @@
 import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.ThreadPoolExecutor;
 import javax.management.MBeanServer;
+
```
Comment (tkalkirill) [inline modules/core/src/test/java/org/apache/ignite/testframework/junits/IgniteTestResources.java:27]: Not needed

**Review by tkalkirill — APPROVED** [review]

**Comment by rpuch** [comment]
Thanks for reviewing and merging!

**Pull request closed without merging**